src/frontend/tsx/GamePanel.tsx

```diff
@@ -22,46 +25,46 @@ export function GamePanel () {
     await pokeAction(urbit, claimSpecialDraw(gameID))
   }
 
+  // from stackoverflow, only slightly modified to satisfy
+  // type checker
+  function sliceIntoChunks(arr: Array<SAN>, chunkSize: number) {
+    const res = [];
+    for (let i = 0; i < arr.length; i += chunkSize) {
+        const chunk = arr.slice(i, i + chunkSize);
+        res.push(chunk);
+    }
+    return res;
+}
+
   return (
     <div className='game-panel-container col'>
       <div className="game-panel col">
-        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' invisible')}>
+        <div id="opp-timer" className={'timer row' + (hasGame ? '' : ' hidden')}>
           <p>00:00</p>
         </div>
-        <div id="opp-player" className={'player row' + (hasGame ? '' : ' invisible')}>
-          <p>~sampel-palnet</p>
+        <div id="opp-player" className={'player row' + (hasGame ? '' : ' hidden')}>
+          <p>{opponent}</p>
         </div>
-        <div className="moves col">
-          <div className="moves-divider"></div>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
-          <p> 00  ply ply</p>
+        <div className={'moves col' + (hasGame ? '' : ' hidden')}>
+        <ol>
+        {
+          Array.from(displayMoves).map((ply, thisIndex, thisArray) => {
+            const nextIndex: number = thisIndex + 1
+            if (thisIndex % 2 === 0) {
+              return (
+              <li>
+                <span>{ply}</span> <span>{(nextIndex > thisArray.length ? '' : thisArray.at(nextIndex))}</span>
```

Comment:
This is still WIP, correct? Current code looks like it only works when both moves of a ply have been made.

Reply:
This is WIP until I request a proper review, yes. Since we're not making it work for the practice board anymore, the only thing left to do before review is get it browsing previous board states.

Right now the move list will correctly show a line like "01. e3" as soon as white has made their first move. If a second move hasn't been made, the conditional logic here will append nothing (`''`); if a second move has been made, it'll append the next move. This applies to the moves at index 0, 2, 4, 6 etc. in the array. There's no other logic around numbering/placing the moves in the UI, the numbering is just a CSS rule.

![image](https://0x0.st/onWi.png)